lib/src/app/allthenews_app.dart

```diff
@@ -13,6 +15,8 @@ class AllTheNewsApp extends StatefulWidget {
 }
 
 class _AllTheNewsAppState extends State<AllTheNewsApp> {
+  final Future<FirebaseApp> _firebaseInitialization = Firebase.initializeApp();
```

Comment:
myślę, że może wstrzyknąć? 🤔 przy jakichś testach ui pewnie wypadałoby to mockować

Reply:
ano dało się

Reply:
jednak zostawię to tak jak jest, żeby wstrzyknąć obiekt, który jest inicjalizowany przy pomocy asynchronicznej metody można użyć `registerFactoryAsync` lub analogicznie singleton, ale właśnie o to chodzi, żeby w tej klasie użyć future buildera do inicjalizacji i pokazać błąd jak coś nie pójdzie

Reply:
chyba, że wyjebać całkiem tą obsługę inicjalizacji FirebaseApp i zostawić to wszystko po stronie get_it, ale zostawiłbym to do sprawdzenia czy okazji testów ui

Reply:
Można to owrapować w klasę FirebaseInitializer czy coś takiego i ona może zwracać Future<FirebaseApp> - wtedy można ją zamockować. Zgadzam się z tym, że obsługę warto zostawić :)